Podcast/Authentication.swift

```diff
@@ -15,10 +15,48 @@ enum AuthenticationEndpointRequestType {
     case merge
 }
 
+enum SignInResult {
+    case success
+    case cancelled
+    case failure
+
+    // to convert from Facebook result
+    static func conversion(from loginResult: LoginResult) -> SignInResult {
+        switch(loginResult) {
+        case .cancelled:
+            return SignInResult.cancelled
+        case .failed:
+            return SignInResult.failure
+        case .success:
+            return SignInResult.success
+        }
+    }
+
+    // to convert from Google error
+    static func conversion(from error: Error?) -> SignInResult {
+        switch(error) {
+        case .none:
+            return SignInResult.success
+        case .some(let googleError):
+            switch(googleError.code) {
+            case GIDSignInErrorCode.canceled.rawValue, GIDSignInErrorCode.hasNoAuthInKeychain.rawValue:
+                return SignInResult.cancelled
+            default:
+                return SignInResult.failure
```

Comment:
not really sure what happened but i added my google account and it just took me to google lol and i couldn't really exit the view unless if i tapped on the bottom tab bar
![img_664efb63a346-1](https://user-images.githubusercontent.com/20246620/37192522-ea489614-2333-11e8-84f7-47622e2ae454.jpeg)

Reply:
Yes I know this is the only current bug :( ... I commented where it is because I think its a Google SDK issue and not our issue.

Reply:
Yea both are very buggy (when I change Facebook from the web sign in behavior)  .. maybe I can ask @dantheli but im struggling with login here. It's an issue with completion blocks not being called and views (such as the one above) not popping off like they should